src/coreComponents/codingUtilities/Utilities.hpp

```diff
@@ -161,6 +161,160 @@ void applyChainRuleInPlace( localIndex const N,
   copy( N, work, df_dxy );
 }
 
+// TODO: make sure this is a good place
+// TODO: check/fix the implementation of these functions
+
+template< typename T >
+void findSurroundingIndex( array1d< T > const & x,
+                           T xval,
+                           integer & iminus,
+                           integer & iplus )
+{
+  GEOSX_THROW_IF( x.empty(), "Interpolation table is empty", InputError );
+  GEOSX_THROW_IF( x[0] > xval, "Input x value is out of range, extrapolation not allowed", InputError );
+
+  // search for interval
+  for( iplus = 1; iplus < x.size() - 1 && x[iplus] < xval; ++iplus )
+  { }
+  iminus = iplus - 1;
+}
+
+
+// TODO: replace with something from LvArray
+template< typename T >
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void findSurroundingIndex( arrayView1d< T const > const & x,
+                           T xval,
+                           integer & iminus,
+                           integer & iplus )
+{
+  // search for interval
+  for( iplus = 1; iplus < x.size() - 1 && x[iplus] < xval; ++iplus )
+  { }
+  iminus = iplus - 1;
+}
+
+template< typename T >
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void findSurroundingIndex( arraySlice1d< const T > const & x,
+                           T xval,
+                           integer & iminus,
+                           integer & iplus )
+{
+  // search for interval
+  for( iplus = 1; iplus < x.size() - 1 && x[iplus] < xval; ++iplus )
+  { }
+  iminus = iplus - 1;
+}
+
+template< typename T >
+void interpolation1( array1d< T > const & xin,
+                     array1d< T > const & yin,
+                     array1d< T > const & xout,
+                     array1d< T > & yout )
+{
+  GEOSX_THROW_IF( xin.size() != yin.size(), "interpolation1: size mismatch!", InputError );
+  GEOSX_THROW_IF( xout.size() != yout.size(), "interpolation1: size mismatch!", InputError );
+  for( localIndex n = 0; n != xout.size(); ++n )
+  {
+    T const x = xout[n];
+    GEOSX_THROW_IF( xin[0] > x, "interpolation1: input x out of range, cannot extrapolate below the limits", InputError );
+
+    // search for interval
+    integer i_minus, i_plus;
+    findSurroundingIndex( xin, x, i_minus, i_plus );
+
+    if( i_minus == i_plus )
+    {
+      yout[n] = yin[i_minus];
+    }
+    else
+    {
+      yout[n] = yin[i_minus] * ( xin[i_plus] - x ) / ( xin[i_plus] - xin[i_minus] )
+                + yin[i_plus] * ( ( x - xin[i_minus] ) / ( xin[i_plus] - xin[i_minus] ) );
+      GEOSX_THROW_IF( yout[n] < 0.0, "interpolation1: negative value", InputError );
```

Comment:
The assumption here is that all `yin` are non-negative and therefore you can't have a negative interpolated value? This is something specific to the call site, not the interpolation routine itself. I recommend checking output values at the call site if desired, or not check them at all.